bark/world/objects/agent.cpp

```diff
@@ -73,7 +74,11 @@ Agent::Agent(const Agent& other_agent)
 
 void Agent::PlanBehavior(const double& min_planning_dt,
                          const ObservedWorld& observed_world) {
+  auto timer = bark::commons::timer::Timer();
+  timer.Start();
   behavior_model_->Plan(min_planning_dt, observed_world);
+  double duration = timer.DurationInSeconds();
+  behavior_model_->SetLastSolutionTime(duration);
```

Comment:
I would rather set the last time the behavior has been planned, and e.g. return a logging message if the planning time was larger then the step time dt (e.g., 0.2s)

Reply:
This would also enable replanning after a specific time

Reply:
Or the behavior model itself could handle the timer as, e.g., the IDM does not need to be timed.

Reply:
the time measurement is rather for the benchmarking, I would say. In the end, we will be able to evaluate pretty easily of the planning is real-time capable or not. Thus, I would prefer to save the time. We could move it to the behavior model (that's what I did for MIQP already, but @juloberno wish was to make this available for every model in a generic way. I don't think that it adds much overhead, but we could introduce a flag in the behavior model, that states wether time should be measured or not. In the end, this flag could be set only for the evaluation agent, what do you think?

Reply:
I think it is a meaningful feature, but the implementation is confusing. 
If the agent measures timing, then the agent should have the planning time. Otherwise the behavior model should handle this internally with a redefinition of the Plan interface and and an additional purely virtual PlanImplementation function.

Also, it should be possible to avoid overhead, e.g. using a parameter. Maybe, then still the behavior-model specific solution is better.

Reply:
what is the status here, should we agree on a solution?

I think it would be interesting to see how long the model runs in general and how efficient it is. Just wanted to note that if we saved a timestamp when the behavior/execution model last planned a trajectory, we could enable replanning after a specific time.

Reply:
I've already implemented something, I will add a test and then push it.